packages/manager/modules/vps/src/dashboard/tile/configuration/service.js

```diff
@@ -96,8 +96,16 @@ export default class ConfigurationTileService {
       };
     }
 
+    const availableUpgradePlancodes = this.availableUpgrades.map(
+      (upgrade) => upgrade.planCode,
+    );
+    // Exclude all products which aren't listed in available upgrades
+    const availableCatalogProducts = this.catalog.products.filter((product) =>
+      availableUpgradePlancodes.includes(product.name),
+    );
```

Comment:
**suggestion:** Since upgrades and catalog has huge list, `availableUpgradePlancodes` can be maintained as an object for easily finding out if the upgrade exists.
```suggestion
    const availableUpgradePlancodes = this.availableUpgrades.reduce(
      (acc, upgrade) => {acc[upgrade.planCode]: true},
      {}
    );
    // Exclude all products which aren't listed in available upgrades
    const availableCatalogProducts = this.catalog.products.filter((product) =>
      availableUpgradePlancodes[product.name],
    );
```